openedx_tagging/core/tagging/api.py

```diff
@@ -136,3 +136,37 @@ def tag_object(
     Preserves existing (valid) tags, adds new (valid) tags, and removes omitted (or invalid) tags.
     """
     return taxonomy.cast().tag_object(tags, object_id)
+
+
+def autocomplete_tags(
+    taxonomy: Taxonomy,
+    search: str,
+    object_id: str = None,
+) -> QuerySet:
+    """
+    Provides auto-complete suggestions by matching the `search` string against existing
+    ObjectTags linked to the given taxonomy. A case-insensitive search is used in order
```

Comment:
I think it's very unexpected that this only matches existing ObjectTags.

I know the rationale: 

> for closed taxonomies, the frontend can handle auto-completion without consulting the backend (e.g using [autocompleter](https://www.npmjs.com/package/autocompleter)), because it's already fetched all of the Tags in a Taxonomy to display the tree of options

and that makes sense, but this could still be confusing.

So I suggest that you either:
1) Put that explanation into the docstring, or
2) Add a new parameter like `applied_tags_only=True`. When it's true, it only returns tags based on the existing objecttag_set (i.e. what you already implemented), and when it's `False` it raises `NotImplementedError("(explanation)...For now we recommend fetching all of the taxonomy's tags using get_tags() and filtering them on the frontend")`

Reply:
Updated [here](https://github.com/openedx/openedx-learning/pull/59/commits/7c30263a4b5774ccb9c018ec9421b5f3d6022018#diff-951a5c4d32e1301ac1c30419ebce4ada8ae17db6c93f4c2a5a34caac9eb30e0aR174)